src/hooks/schemas.tsx

```diff
@@ -90,3 +90,39 @@ export function useCreateSchema({
     },
   });
 }
+
+export interface DeleteSchemaMutationParams
+  extends Pick<SchemaInterface, 'catalog_name' | 'name'> {}
+
+interface DeleteSchemaParams {
+  onSuccessCallback?: () => void;
+  catalog: string;
+}
+
+export function useDeleteSchema({
+  onSuccessCallback,
```

Comment:
Sorry I just captured this.
You don't need to do create a new prop here to handle on success.

You can just simply do onSuccess in the caller
```tsx
mutation.mutate(values, {
  onSuccess: () => {}
});
```
Can we have follow-up PRs to remove this pattern on other deletion hooks?

Reply:
Do you mean to move the `invalidateQueries` to the caller as well?

Reply:
Similar example: https://github.com/TanStack/query/blob/1fc6124bfd22a26095ab1fca20caa39b4036fd86/examples/react/auto-refetching/src/pages/index.tsx#L39

Reply:
Ok got it, but we don't need to pass in `onSuccessCallback`. I will do a follow-on PR to remove all those. Thanks @yc-shawn 👍